lib/spack/spack/test/build_types.py

```diff
@@ -0,0 +1,33 @@
+# Copyright 2013-2021 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+import inspect
+
+import spack.build_types
+from spack.util.naming import class_to_mod
+
+
+def test_class_to_mod():
+    """
+    Ensure that every module in build_types has a key via class_to_mod.
+    """
+    classes = inspect.getmembers(spack.build_types, inspect.isclass)
+    for classname, _ in classes:
+
+        # This is more of an abstract class
+        if classname == "BuildTypeBase":
+            continue
+
+        build_type = class_to_mod(classname)
+        assert build_type in spack.build_types.build_types
```

Comment:
Someday we should move this and the tests for `mod_to_class` to their own test file `lib/spack/spack/test/util/naming.py`. But that probably doesn't have to be now.

Reply:
I went there to look for it, didn't see the file, then grepped for mod_to_class, saw it was alongside package.py, so figured since I needed to eventually add build_types.py in the test module I might as well put it there! I agree we should eventually refactor to test what is in util/naming.py in the matching test file.